src/main/java/com/yo1000/heagent/HealthCheckAgent.java

```diff
@@ -47,9 +52,16 @@ public static Server createServer(String agentArgs) {
                         })
                         .collect(Collectors.joining("/")));
 
-        Server server = hostname != null && !hostname.equals("0.0.0.0")
-                ? new Server(new InetSocketAddress(hostname, port))
-                : new Server(port);
+
+        QueuedThreadPool queuedThreadPool = new QueuedThreadPool();
+        queuedThreadPool.setDaemon(true);
+        Server server = new Server(queuedThreadPool);
+
+        ScheduledExecutorScheduler scheduledExecutorScheduler = new ScheduledExecutorScheduler("Connector-Scheduler-Daemonized", true);
+        ServerConnector connector = new ServerConnector(server, null, scheduledExecutorScheduler, null, -1, -1, new HttpConnectionFactory());
+        connector.setHost(hostname != null ? hostname : "127.0.0.1");
```

Comment:
This agent is used for health checks, so if only self-host is set as the bind address, it will look like it cannot be monitored.

```suggestion
        connector.setHost(hostname != null ? hostname : "0.0.0.0");
```